packages/vega-parser/src/parsers/guides/axis-util.js

```diff
@@ -0,0 +1,47 @@
+import { stringValue } from 'vega-util';
+import { Bottom, Left, Top } from './constants';
+import { isSignal } from '../../util';
+
+export function xyAxisSignalRef(xy, axisOrientExpr, yes, no) {
+  var yesStr = isSignal(yes) ? yes.signal : stringValue(cleanValue(yes));
+  var noStr = isSignal(no) ? no.signal : stringValue(cleanValue(no));
+  return {
+    signal: `${xyAxisBooleanExpr(xy, axisOrientExpr)} ? (${yesStr}) : (${noStr})`
+  };
+}
+  
+export function xyAxisBooleanExpr(xy, axisOrientExpr) {
+  return `${xy === 'x' ? '' : '!'}((${axisOrientExpr}) === "${Top}" || (${axisOrientExpr}) === "${Bottom}")`;
```

Comment:
Referencing expressions multiple times like this just inlines the full expression, leading to redundant evaluation. Obviously the expression language doesn't support defining variables, but another option could be to define an encoding property once (either on an item or parent item) and then reference it in the formula. I don't think it is critical to do this now, but something to consider.

Reply:
At least for this function, we could use `indexOf(["${Top}", "${Bottom}"], ${axisOrientExpr}) >= 0` to call the axisOrientExpr only once here. 

Of course, we still have redundant calculation across calls of `xyAxisBooleanExpr`, but at least we will reduce the redundancy by half.

Reply:
Switched to `indexof` for now.

Reply:
I've also reduce the redundancy of cleaning values (for `signalRefs` and `undefined`).